sdk/core/azure-core/inc/azure/core/internal/http/user_agent.hpp

```diff
@@ -10,26 +10,35 @@
 
 #include <string>
 
-namespace Azure { namespace Core { namespace Http { namespace _detail {
-  // NOTE: Treat Azure::Core::Http::_detail::UserAgentGenerator::GenerateUserAgent() as _internal -
-  // it is being/has been used by eventhubs.
+namespace Azure { namespace Core { namespace Http { namespace _internal {
+  /**
+   * @brief Telemetry User-Agent string generator.
+   *
+   */
   class UserAgentGenerator {
   public:
+    /**
+     * @brief Generates User-Agent string for telemetry.
+     *
+     * @param componentName the name of the SDK component.
+     * @param componentVersion the version of the SDK component.
+     * @param applicationId user application ID
+     * @param cplusplusValue value of the `__cplusplus` macro.
```

Comment:
For my edification, why do we need upstream callers (client SDK packages like EventHubs) to pass in this value of `cplusplusValue` as a parameter, explicitly?

Why not have the user-agent implementation in core be the only place where we fetch/read the `__cplusplus` macro value, and set it? This makes the caller's job easy. Do we expect per SDK package differences for this value?

Reply:
Oh yes, I remember, we have option fields to set them to different values that isn't tied to the language version.

I didn't realize we ended up using private members on the options struct (with friend) to set it, per package:
https://github.com/Azure/azure-sdk-for-cpp/blob/0e48f058d420b205dbff93a84c49a2d23ad52e62/sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp#L77-L93